src/librustc/ty/mod.rs

```diff
@@ -555,16 +555,29 @@ impl<'tcx> Hash for TyS<'tcx> {
 impl<'tcx> TyS<'tcx> {
```

Comment:
Looks like all this impl is used only for diagnostics.
Could it be moved into some diagnostic-related file / module?

Having comments on functions like `is_simple_ty` would also be nice because the name alone leaves so many questions.